Sum type in main() -- C compilation error (from tcc)

### Describe the bug

works in module

![Image](https://github.com/user-attachments/assets/1491b03c-b4e6-485a-99d8-268d7e869ac9)

but do not works in main()

![Image](https://github.com/user-attachments/assets/1d4b31b4-3fc0-45d8-916a-709fbd4982ea)

Terminal output
```
PS D:\OSPanel\domains\yp\v-helloworld> v run .
string
PS D:\OSPanel\domains\yp\v-helloworld> v run .
test.Dbv
PS D:\OSPanel\domains\yp\v-helloworld> v run .
test.Dbv
string
PS D:\OSPanel\domains\yp\v-helloworld> v run .
================== C compilation error (from tcc): ==============
cc: C:/Users/vodis/AppData/Local/Temp/v_0/v-helloworld.01JM5D0KTAQFRBYTX54F0JTJTQ.tmp.c:2097: warning: implicit declaration of function 'charptr_vstring_literal'
cc: C:/Users/vodis/AppData/Local/Temp/v_0/v-helloworld.01JM5D0KTAQFRBYTX54F0JTJTQ.tmp.c:2097: error: cannot convert 'int' to 'struct string'
... (the original output was 4 lines long, and was truncated to 2 lines)
=================================================================
(You can pass `-cg`, or `-show-c-output` as well, to print all the C error messages).
builder error:
==================
C error found. It should never happen, when compiling pure V code.
This is a V compiler bug, please report it using `v bug file.v`,
or goto https://github.com/vlang/v/issues/new/choose .
You can also use #help on Discord: https://discord.gg/vlang .
PS D:\OSPanel\domains\yp\v-helloworld> v version
V 0.4.9 92f436d
```

### Reproduction Steps
```v
module main

// import test

type Dbv = int | string

fn main() {
	// test.t_1()

	mut m := Dbv('sd')
	// m = 'sdvsdv1'

	println(typeof(m).name)
	println(m.type_name())
}
```
### Expected Behavior

screen 1 

### Current Behavior

screen 2

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

V 0.4.9 92f436d

### Environment details (OS name and version, etc.)

Windows 11
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/ast/table.v
+++ b/vlib/v/ast/table.v
@@ -28,6 +28,7 @@ pub mut:
 	arr_reverse      bool            // arr.reverse()
 	arr_init         bool            // [1, 2, 3]
 	arr_map          bool            // []map[key]value
+	type_name        bool            // var.type_name()
 	map_update       bool            // {...foo}
 	interpolation    bool            // '${foo} ${bar}'
 	option_or_result bool            // has panic call

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -1988,6 +1988,7 @@ fn (mut c Checker) method_call(mut node ast.CallExpr, mut continue_check &bool) 
 	}
 	if left_sym.kind in [.sum_type, .interface] {
 		if method_name == 'type_name' {
+			c.table.used_features.type_name = true
 			return ast.string_type
 		}
 		if method_name == 'type_idx' {

--- a/vlib/v/markused/markused.v
+++ b/vlib/v/markused/markused.v
@@ -212,6 +212,9 @@ pub fn mark_used(mut table ast.Table, mut pref_ pref.Preferences, ast_files []&a
 			core_fns << '__print_assert_failure'
 			core_fns << 'isnil'
 		}
+		if table.used_features.type_name {
+			core_fns << charptr_idx_str + '.vstring_literal'
+		}
 		if pref_.trace_calls || pref_.trace_fns.len > 0 {
 			include_panic_deps = true
 			core_fns << 'vgettid'

--- a/vlib/v/tests/skip_unused/sumtype_type_name.run.out
+++ b/vlib/v/tests/skip_unused/sumtype_type_name.run.out
@@ -1,0 +1,2 @@
+Dbv
+string

--- a/vlib/v/tests/skip_unused/sumtype_type_name.skip_unused.run.out
+++ b/vlib/v/tests/skip_unused/sumtype_type_name.skip_unused.run.out
@@ -1,0 +1,2 @@
+Dbv
+string

--- a/vlib/v/tests/skip_unused/sumtype_type_name.vv
+++ b/vlib/v/tests/skip_unused/sumtype_type_name.vv
@@ -1,0 +1,9 @@
+module main
+
+type Dbv = int | string
+
+fn main() {
+	mut m := Dbv('foo')
+	println(typeof(m).name)
+	println(m.type_name())
+}

